File: HW3/MultitouchCanvasViewBoilerPlate.java
package edu.ucsb.cs.cs184.holl.thollerermmmedley;

import android.content.Context;
import android.content.res.TypedArray;
import android.graphics.Bitmap;
import android.graphics.Canvas;
import android.graphics.Color;
import android.graphics.Paint;
import android.graphics.PointF;
import android.util.AttributeSet;
import android.view.MotionEvent;
import android.view.SurfaceHolder;
import android.view.SurfaceView;
import androidx.core.view.MotionEventCompat;
import java.util.HashMap;
import java.util.LinkedList;
import java.util.Map;
import java.util.Queue;

public class MultitouchCanvasView extends SurfaceView implements SurfaceHolder.Callback {

    public MultitouchCanvasView(Context context, AttributeSet attributes) {
        super(context, attributes);
        getHolder().addCallback(this);
    }

    @Override
    public void surfaceCreated(SurfaceHolder holder) {
        //Called very first after constructor
    }

    @Override
    public void surfaceChanged(SurfaceHolder holder, int format, int width, int height) {
    }

    @Override
    public void surfaceDestroyed(SurfaceHolder holder) {
        //Called when orientation changes
    }

    @Override
    public boolean onTouchEvent(MotionEvent event) {
        //Handle MotionEvent types here
        return true;
    }
}

File: HW3/MultitouchCanvasViewBoilerPlate.kt
package com.apps.guida.cs184_f19_hw3_solution.ui.multitouch

import android.content.Context
import android.content.res.TypedArray
import android.graphics.*
import android.util.AttributeSet
import android.view.MotionEvent
import android.view.SurfaceHolder
import android.view.SurfaceView
import java.util.*
import android.icu.lang.UCharacter.GraphemeClusterBreak.T
import kotlin.collections.HashMap
import android.icu.lang.UCharacter.GraphemeClusterBreak.T
import android.util.Log
import android.util.TypedValue
import androidx.core.view.MotionEventCompat
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModelProviders
import com.apps.guida.cs184_f19_hw3_solution.R

class MultitouchCanvasView(context: Context, attr: AttributeSet): SurfaceView(context, attr) , SurfaceHolder.Callback {

    init {
        //Initialize member variables
        Log.d("MULTITOUCH", "CanvasView: Constructor")
        holder.addCallback(this)
    }

    override fun surfaceCreated(holder: SurfaceHolder?) {
        //Called very first after constructor
        Log.d("MULTITOUCH", "CanvasView: surfaceCreated")
    }
    override fun surfaceChanged(holder: SurfaceHolder?, format: Int, width: Int, height: Int) {
        Log.d("MULTITOUCH", "CanvasView: surfaceChanged")
    }

    override fun surfaceDestroyed(holder: SurfaceHolder?) {
        //Called when screen rotates
        Log.d("MULTITOUCH", "CanvasView: surfaceDestroyed")
    }

    override fun onTouchEvent(event: MotionEvent?): Boolean {
        //Handle the different types of MotionEvents here
        Log.d("MULTITOUCH", "CanvasView: onTouchEvent")
        return true
    }
}

